### PR TITLE
Fix a few issues

### PR DESCRIPTION
**Pull request by mgalloue**
Typo: Fix issue on brush badly resizing in a particular case (with only NAN values on a dimension).
Typo: Fix issue on drag axis (available width doesn't exit, use width instead)
Update availableHeight and availableWidth function to avoid negative values (use zero instead).

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -598,14 +598,14 @@ nv.utils.sanitizeWidth = function(width, container) {
 Calculate the available height for a chart.
 */
 nv.utils.availableHeight = function(height, container, margin) {
-    return nv.utils.sanitizeHeight(height, container) - margin.top - margin.bottom;
+    return Math.max(0,nv.utils.sanitizeHeight(height, container) - margin.top - margin.bottom);
```
Comment (liquidpele) [inline src/utils.js]: why is this necessary?

File: src/models/parallelCoordinates.js [hunk 2]
```diff
@@ -366,7 +377,7 @@ nv.models.parallelCoordinates = function() {
             }
 
             function dragMove(d) {
-                dragging[d.key] = Math.min(availableWidth, Math.max(0, this.parentNode.__origin__ += d3.event.x));
+                dragging[d.key] = Math.min(width, Math.max(0, this.parentNode.__origin__ += d3.event.x));
```
Comment (liquidpele) [inline src/models/parallelCoordinates.js]: width is only set if they use that option, otherwise it's null, why would you want this change?

File: src/models/parallelCoordinates.js [hunk 3]
```diff
@@ -100,7 +100,7 @@ nv.models.parallelCoordinates = function() {
                         max = y[d].domain()[1];
                     }
                         //If the brush extent is > max (< min), keep the extent value.
-                    else if (!f[0].hasNaN && displayBrush) {
+                    else if (!f[0].hasOnlyNaN && displayBrush) {
```
Comment (liquidpele) [inline src/models/parallelCoordinates.js]: What is hasOnlyNaN?   Does this replace hasNaN, or a supplement to it?

File: src/utils.js (hunk 1 above)
Comment (mgalloue) [inline src/utils.js]: Because some time sanitizeWidth return zero and with the margin, availableWidth could return a negative value giving js error. So if sanitizeWidth is zero then available width too.

File: src/models/parallelCoordinates.js (hunk 3 above)
Comment (mgalloue) [inline src/models/parallelCoordinates.js]: HasOnlyNan is already existing, in fact when I commit this code in a first place I made a typo on setting hasNaN, here we need to check hasOnlyNaN. This check if there is only nan values on the brush (so the brush is below the dimension axis, and on the undefined values line).

File: src/models/parallelCoordinates.js (hunk 2 above)
Comment (mgalloue) [inline src/models/parallelCoordinates.js]: On my case it doesn't work with availableWidth and works well with width but your comment make sense. I will try to understand why so I removed this from the PR for now

File: src/tooltip.js [hunk 4]
```diff
@@ -47,8 +47,8 @@ nv.models.tooltip = function() {
      */
     var position = function() {
         return {
-            left: d3.event.clientX,
-            top: d3.event.clientY
+            left: d3.event !== null ? d3.event.clientX : 0,
+            top: d3.event !== null ? d3.event.clientY : 0
```
Comment (mgalloue) [inline src/tooltip.js]: This is needed because when data are updated, if the tooltip is currently display, it can stay visible even if the mouse is not on the point anymore, because no mouse move event is detected. Later if I go out of the chart the tooltip is still displaying (and never hide until going on another point). To avoid that I called chart.tooltip.hidden(true) at each update, but d3.event is null in this case.
